api/serializers.py

```diff
@@ -1,36 +1,88 @@
 
+    name = serializers.CharField(read_only=True, label="Product name")
 
-class CreateCommissionSerializer(serializers.ModelSerializer):
-    name = serializers.CharField(required=True,
-                                 validators=[UniqueValidator(queryset=Commission.objects.all(),
-                                                             message='Commission already exists.')])
+    price = serializers.IntegerField(read_only=True, label="Product price in NOK")
 
-    class Meta:
-        model = Commission
-        fields = '__all__'
+    description = serializers.CharField(read_only=True, allow_blank=True, allow_null=True, label="Product description",
+                                        help_text="Returns `null` if no description exists")
 
+    icon = serializers.CharField(read_only=True, label="Product icon descriptor")
 
-# ===============================
-# USERS
-# ===============================
 
+class CheckBalanceSerializer(serializers.Serializer):
+    user = serializers.CharField(read_only=True, label="User´s full name")
+
+    balance = serializers.IntegerField(source='public_balance', read_only=True, allow_null=True, label="Balance in NOK",
+                                       help_text="Returns `null` if user has disabled public display of balance")
+
+    has_sufficient_funds = serializers.BooleanField(read_only=True,
+                                                    label="Whether the user has enough funds left to charge or not")
+
+
+class ChargeSociBankAccountDeserializer(serializers.Serializer):
+    sku = serializers.CharField(label="Product SKU number to charge for")
+
+    amount = serializers.IntegerField(required=False, label="Amount to charge",
+                                      help_text="Only required when using the SKU for charging a direct amount")
+
+    @staticmethod
+    def validate_sku(value):
+        if not SociProduct.objects.filter(sku_number=value).exists():
+            raise serializers.ValidationError('SKU number is invalid.')
+        return value
+
+    @staticmethod
+    def validate_amount(value):
+        if value <= 0:
+            raise serializers.ValidationError('Amount must be positive.')
+        return value
+
+    def validate(self, attrs):
+        if attrs.get('sku', None) != settings.DIRECT_CHARGE_SKU:
+            if attrs.get('amount', None):
+                raise serializers.ValidationError('Amount can only be set when charging a direct amount.')
+
+            attrs['amount'] = SociProduct.objects.get(sku_number=attrs['sku']).price
+
+        elif attrs.get('sku', None) == settings.DIRECT_CHARGE_SKU and attrs.get('amount', None) is None:
+            raise serializers.ValidationError('Amount must be set when charging a direct amount.')
+
+        if attrs['amount'] > self.instance.balance:
+            raise InsufficientFundsException()
+
+        return attrs
+
+    def update(self, instance: SociBankAccount, validated_data):
```

Comment:
Should it be possible to "kryss" more than one of a product at once?

Reply:
Great question! (Read: Oops, didn't think about that...)